### PR TITLE
Azure Monitor Exporter - Add Exception Telemetry

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
This is based on Application Insights. Borrowed few implementation from dotnet Application Insights repo.

**Example**

```json
{"name":"Exception","time":"2022-01-29T02:31:48.7875987Z","iKey":"Ikey","tags":{"ai.internal.sdkVersion":"dotnet4.8.4420.0:otel1.2.0-rc1:ext1.0.0-alpha.20220129.1","ai.cloud.role":"my-namespace.my-service","ai.cloud.roleInstance":"my-instance"},"data":{"baseType":"ExceptionData","baseData":{"exceptions":[{"id":12547953,"typeName":"System.InvalidOperationException","message":"My test message","hasFullStack":true,"parsedStack":[{"level":0,"method":"Azure.Monitor.OpenTelemetry.Exporter.Demo.Tracing.DemoTrace.CallFun1","assembly":"Azure.Monitor.OpenTelemetry.Exporter.Demo.Tracing, Version=1.0.0.0, Culture=neutral, PublicKeyToken=6a281e58f0dfecb7"},{"level":1,"method":"Azure.Monitor.OpenTelemetry.Exporter.Demo.Tracing.DemoTrace.Main","assembly":"Azure.Monitor.OpenTelemetry.Exporter.Demo.Tracing, Version=1.0.0.0, Culture=neutral, PublicKeyToken=6a281e58f0dfecb7"}]}],"severityLevel":"Error","problemId":"System.InvalidOperationException at Azure.Monitor.OpenTelemetry.Exporter.Demo.Tracing.DemoTrace.Main:253","ver":2}}}
```

**TODO:**

* Add test case
* Add Demo project that covers both logs and exception

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryExceptionData.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    internal partial class TelemetryExceptionData
+    {
+        internal const int MaxExceptionCountToSave = 10;
+
+        public TelemetryExceptionData(int version, LogRecord logRecord) : base(version)
+        {
+            SeverityLevel = LogsHelper.GetSeverityLevel(logRecord.LogLevel);
+            ProblemId = LogsHelper.GetProblemId(logRecord.Exception);
+
+            var message = logRecord.State.ToString();
```
Comment (cijothomas) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryExceptionData.cs:21]: `State` could be null

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/LogsHelper.cs [hunk 2]
```diff
@@ -23,17 +27,62 @@ internal static List<TelemetryItem> OtelToAzureMonitorLogs(Batch<LogRecord> batc
                 telemetryItem = new TelemetryItem(logRecord);
                 telemetryItem.InstrumentationKey = instrumentationKey;
                 telemetryItem.SetResource(roleName, roleInstance);
-                telemetryItem.Data = new MonitorBase
+                if (logRecord?.Exception != null)
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/LogsHelper.cs:30]: nit: logrecord will not be null.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryExceptionData.cs (hunk 1 above)
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/TelemetryExceptionData.cs:21]: Should we look at `logrecord.FormattedMessage` first ( if `IncludeFormattedMessage=true`) and then `logrecord.State` (if `ParseStateValues=false`) and if that also fails look at `logrecord.StateValues` {OriginalFormat} ?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/InnerExceptionCountExceededException.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Serialization;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter
+{
+    /// <summary>
+    /// This exception is used to notify the user that the set of inner exceptions has been trimmed because it exceeded our allowed send limit.
+    /// </summary>
+    [Serializable]
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/InnerExceptionCountExceededException.cs:13]: do we need this?

**Review by vishweshbankwar — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/InnerExceptionCountExceededException.cs (hunk 3 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/InnerExceptionCountExceededException.cs:13]: Will get this removed, at this point there is no need for this attribute.

**Pull request merged**